programs/pkey/gen_key.c

```diff
@@ -173,15 +179,35 @@ static int write_private_key( mbedtls_pk_context *key, const char *output_file )
     memset(output_buf, 0, 16000);
     if( opt.format == FORMAT_PEM )
     {
-        if( ( ret = mbedtls_pk_write_key_pem( key, output_buf, 16000 ) ) != 0 )
-            return( ret );
+        if( opt.syntax == SYNTAX_PKCS1 )
+        {
+            ret = mbedtls_pk_write_key_pem( key, output_buf, 16000 );
```

Comment:
preexisitng: I think it would be a good idea to change 16000 to a precompilation MACRO definiiton

Reply:
Ok